frostsnapp/native/src/api.rs

```diff
@@ -647,16 +649,18 @@ impl Wallet {
         let _enter = span.enter();
         let start = Instant::now();
         event!(TLevel::INFO, "starting sync");
-        let mut sync_request = {
+        let sync_request = {
+            let inspect_stream = stream.clone();
             let wallet = self.inner.lock().unwrap();
-            wallet.start_sync(key_id)
+            let sync_req = wallet.start_sync(key_id);
+            let total = sync_req.spks.len();
+            let mut i = 0;
+            sync_req.inspect_spks(move |_spk| {
+                inspect_stream.add((i as f64) / (total - 1) as f64);
```

Comment:
On this progress of syncing spks, there can never be just one spk?

Reply:
This does indeed cause a /0 error if there's only one receive address in the graph.

Reply:
Yes it should be +1 to the numerator rather than -1 from the denominator.

Reply:
961f801cc76f98af994df0135e8bf952c6767607